src/core/lib/security/credentials/fake/fake_credentials.c

```diff
@@ -35,20 +35,90 @@
 
 #include <string.h>
 
-#include "src/core/lib/channel/channel_args.h"
-#include "src/core/lib/iomgr/executor.h"
-
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
 
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/iomgr/executor.h"
+#include "src/core/lib/security/transport/secure_naming.h"
+#include "src/core/lib/support/string.h"
+
 /* -- Fake transport security credentials. -- */
 
+static bool check_target(const char *target_type, const char *target,
+                         const char *set_str) {
+  GPR_ASSERT(target_type != NULL);
+  GPR_ASSERT(target != NULL);
+  char **set = NULL;
+  size_t set_size = 0;
+  gpr_string_split(set_str, ",", &set, &set_size);
+  bool found = false;
+  for (size_t i = 0; i < set_size; ++i) {
+    if (set[i] != NULL && strcmp(target, set[i]) == 0) found = true;
+  }
+  for (size_t i = 0; i < set_size; ++i) {
+    gpr_free(set[i]);
+  }
+  gpr_free(set);
+  return found;
+}
+
+static bool is_lb_channel(const grpc_channel_args *args) {
+  return grpc_glb_targets_info_find_in_args(args) != NULL;
+}
+
 static grpc_security_status fake_transport_security_create_security_connector(
     grpc_exec_ctx *exec_ctx, grpc_channel_credentials *c,
     grpc_call_credentials *call_creds, const char *target,
     const grpc_channel_args *args, grpc_channel_security_connector **sc,
     grpc_channel_args **new_args) {
+  const grpc_arg *expected_target_arg =
```

Comment:
Based on [previous comments](https://github.com/grpc/grpc/pull/9194#discussion_r98025749) the idea was to instantiate the security connector at subchannel creation time, which allowed us to use the appropriate target name for the security connector. This would mean that `check_peer` can "do its thing" without knowing about mappings etc. 

As you recall, a previous commit was doing the checking in the fake security connector's `check_peer`. It could be performed there, but given that the `target` argument passed to the `*_security_create_security_connector` constructors is the definitive one, it can readily be performed here too (unless you feel that doing it in `check_peer` is better as, perhaps, an example).